src/openforms/submissions/attachments.py

```diff
@@ -141,30 +141,22 @@ def validate_uploads(submission_step: SubmissionStep, data: Optional[dict]) -> N
         else:
             allowed_mime_types = glom(component, "file.type", default=[])
 
+        validate_file = mime_type_validator(allowed_mime_types)
```

Comment:
see earlier remark about the class-based validator implementation while still re-using the actual validation code. That should make the code more straightforward to read and maintain.